### PR TITLE
Fluid/shared pointer update and cleanup

### PR DESCRIPTION
**Pull request by jcotela**
Fixing warnings and applying changes in relation to #1328

### CODE REVIEW

**Comment by loumalouomega** [comment]
Sorry @jcotela I put wrongly the ID of the issue and I mentioned you

**Review by RiccardoRossi — APPROVED** [review]
the only suggestion is to pass the clang-tidy right before merging. your call though...

**Review by RiccardoRossi — APPROVED** [review]

**Pull request merged**
